### PR TITLE
Error on invalid yml

### PR DESCRIPTION
**Pull request by lightsofapollo-taskcluster**

### CODE REVIEW

File: src/config/test.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ treeherder:
...
 
 try:
+  errorTaskUrl: "{{{host}}}{{{path}}}/raw-file/{{revision}}/error.yml"
```
Comment (rail) [inline src/config/test.yml]: If I'm reading the above correctly, the error.yml files will be in the root directory of a repo. Should it live closer to other task YAML files? Or maybe in the corresponding product directories (browser, mobile, b2g, etc)?

File: src/jobs/taskcluster_graph.js [hunk 2]
```diff
@@ -29,6 +36,32 @@ function parseUrl(url) {
...
 
+/**
+Fetch a task graph from a url (retires included...)
+*/
+async function fetchGraph(job, url) {
+  assert(url, 'url is required');
+  job.log(`fetching graph ${url}`);
+  let currentRetry = 0;
+  while (currentRetry++ < GRAPH_RETIRES) {
+    try {
```
Comment (rail) [inline src/jobs/taskcluster_graph.js]: This loop is a great candidate for a generic retry() function. Or some existing module like https://github.com/FGRibreau/node-request-retry. I'll be much easier to read, and I bet this pattern will be used somewhere else.

File: src/config/test.yml (hunk 1 above)
Comment (lightsofapollo-taskcluster) [inline src/config/test.yml]: ^ so this config file test.yml is _only_ used in tests (for mozilla-taskcluster)... I actually hardcoded the error graph to something on github until I land the bits to mozilla-central onward.

File: src/jobs/taskcluster_graph.js [hunk 3]
```diff
@@ -67,7 +91,23 @@ export default class TaskclusterGraphJob extends Base {
...
+      job.log("Error creating graph due to yaml syntax errors...");
+      // Even though we won't end up doing anything overly useful we still need
+      // to convey some status to the end user ... The instantiate error should
+      // be safe to pass as it is simply some yaml error.
+      let errorGraphUrl =
+        mustache.render(this.config.try.errorTaskUrl, urlVariables);
+      let errorGraph = await fetchGraph(job, errorGraphUrl);
+      graph = instantiate(errorGraph, Object.assign(variables, {
+        error: e.stack
+      }));
```
Comment (rail) [inline src/jobs/taskcluster_graph.js]: Do we care if the "catch" block fails?

File: src/jobs/taskcluster_graph.js (hunk 2 above)
Comment (lightsofapollo-taskcluster) [inline src/jobs/taskcluster_graph.js]: yeah +1 : I think we are doing this pretty frequently at minimum I will extract this to a util in mozilla-tc

**Comment by lightsofapollo-taskcluster** [comment]
landed in #12

**Pull request closed without merging**